mypy/config_parser.py

```diff
@@ -121,14 +163,29 @@ def parse_config_file(options: Options, set_strict_flags: Callable[[], None],
     else:
         config_files = tuple(map(os.path.expanduser, defaults.CONFIG_FILES))
 
-    parser = configparser.RawConfigParser()
+    config_parser = configparser.RawConfigParser()
 
     for config_file in config_files:
         if not os.path.exists(config_file):
             continue
         try:
-            parser.read(config_file)
-        except configparser.Error as err:
+            if config_file.endswith('.toml'):
+                toml_data = toml.load(config_file, _dict=OrderedDict)
+                # Filter down to just mypy relevant toml keys
+                toml_data = toml_data.get('tool', {})
+                other_keys = [key for key in toml_data.keys()
+                              if key.split('-')[0] != 'mypy']
+                for key in other_keys:
+                    del toml_data[key]
+                if 'mypy' not in toml_data:
+                    continue
+                parser = destructure_overrides(toml_data)
+                config_types = toml_config_types
+            else:
+                config_parser.read(config_file)
+                parser = config_parser
+                config_types = ini_config_types
+        except (toml.TomlDecodeError, configparser.Error, ValueError) as err:
```

Comment:
As mentioned elsewhere, it seems better to use a dedicated exception instead of `ValueError`, since it's possible to get a `ValueError` from a bug in the implementation, and in that case I'd rather get a traceback.